src/pybind/mgr/dashboard/controllers/nvmeof.py

```diff
@@ -22,8 +22,9 @@ class NvmeofNamespace(RESTController):
                          'subsystem_nqn': (str, 'NVMeoF subsystem NQN')
                      })
         def list(self, subsystem_nqn: str):
-            response = MessageToJson(NVMeoFClient().list_namespaces(subsystem_nqn))
-            return json.loads(response)
+            with NVMeoFClient() as client:
+                response = MessageToJson(client.list_namespaces(subsystem_nqn))
```

Comment:
Right now this syntax has changed. Now it's doing:
```py
NVMeoFClient().stub.<grpc_method>(
  NVMeoFClient.pb2.  # doesn't require instantiation since .pb2 is a class variable
    <pb_req>(
      *args,
      **kwargs
    )
  )
```

So one possibility is that `NVMeoF().stub` is a `@property` that accesses a class variable (self._stub) where connections are stored:
```py
class NVMeoFClient:
  pb2 = pb2
  _stub = {}  # because we might want to keep multiple connections to multiple GWs

  @property
  def stub(self):
    if not self._stub:
       # establish new connection
       for gw in NvmeofGatewaysConfig.get_gateways_config()["gateways"].values():
          self._stub[gw] = grpc.insecure_channel(self.gateway_addr)

    # pick one random GW from the list
    return random.choice(list(self._stub.values()))
```
The above only covers connection establishment, but not failures, reconnections, etc.